syntax = "proto3";

package ozonmp.bss_company_api.v1;

import "validate/validate.proto";
import "google/api/annotations.proto";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/ozonmp/bss-company-api/pkg/bss-company-api;bss_company_api";

// BssCompanyApiService - Service for working with companys
service BssCompanyApiService {
  // DescribecompanyV1 - Describe a company
  rpc DescribeCompanyV1(DescribeCompanyV1Request) returns (DescribeCompanyV1Response) {
    option (google.api.http) = {
      get: "/v1/companys/{company_id}"
    };
  }
}

message Company {
  uint64 id  = 1;
  string name = 2;
  string address = 3;
  google.protobuf.Timestamp created = 4;
}

message DescribeCompanyV1Request {
  uint64 company_id = 1 [(validate.rules).uint64.gt = 0];
}

message DescribeCompanyV1Response {
  Company value = 1;
}
